rust/integration-tests/src/sdk/sdk_test_context.rs

```diff
@@ -0,0 +1,165 @@
+use aptos_protos::{indexer::v1::TransactionsResponse, transaction::v1::Transaction};
+use diesel::{pg::PgConnection, Connection};
+use processor::utils::database::new_db_pool;
+use sdk_processor::{config::processor_config, processors::events_processor::ProcessorTrait};
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    ContainerAsync, GenericImage, ImageExt,
+};
+use aptos_indexer_processor_sdk::aptos_indexer_transaction_stream::TransactionStreamConfig;
+use sdk_processor::{
+    config::{
+        db_config::{DbConfig, PostgresConfig},
+    },
+    schema::events::dsl::*,
+};use url::Url;
+
+pub struct SdkTestContext {
+    pub transaction_batches: Vec<Transaction>,
+    postgres_container: ContainerAsync<GenericImage>,
+}
+
+impl SdkTestContext {
+    pub async fn new(txn_bytes: &[&[u8]]) -> anyhow::Result<Self> {
+        // creating new
+        println!("sdk test context new");
+        let transaction_batches = txn_bytes
+            .iter()
+            .map(|txn| {
+                let txn: Transaction = serde_json::from_slice(txn).unwrap();
+                txn
+            })
+            .collect::<Vec<Transaction>>();
+
+        // Set up Postgres container
+        let postgres_container = GenericImage::new("postgres", "14")
+            .with_exposed_port(5432.tcp())
+            .with_wait_for(WaitFor::message_on_stderr(
+                "database system is ready to accept connections",
+            ))
+            .with_env_var("POSTGRES_DB", "postgres")
+            .with_env_var("POSTGRES_USER", "postgres")
+            .with_env_var("POSTGRES_PASSWORD", "postgres")
+            .start()
+            .await
+            .expect("Postgres started");
+
+        Ok(SdkTestContext {
+            transaction_batches,
+            postgres_container,
+        })
+    }
+
+    pub async fn get_db_url(&self) -> String {
+        let host = self.postgres_container.get_host().await.unwrap();
+        let port = self
+            .postgres_container
+            .get_host_port_ipv4(5432)
+            .await
+            .unwrap();
+        format!("postgres://postgres:postgres@{host}:{port}/postgres")
+    }
+
+    /// Helper function to set up and run the mock GRPC server.
+    async fn setup_mock_grpc(&self, transactions: Vec<TransactionsResponse>, chain_id: u64) {
+        println!("received transactions size: {:?}", transactions.len());
+        let mock_grpc_server = MockGrpcServer {
+            transactions,
+            chain_id,
+        };
+
+        // Start the Mock GRPC server
+        tokio::spawn(async move {
+            println!("Starting Mock GRPC server");
+            mock_grpc_server.run().await;
+        });
+    }
+
+    /// Helper function to create TransactionStreamConfig and DbConfig, now part of SdkTestContext.
+    pub async fn create_transaction_and_db_config(
+        &self,
+        starting_version: Option<u64>,
+        ending_version: Option<u64>,
```

Comment:
If there's non-sequential transactions, how would this work? I guess we could make this take in a vec of transactions and get the first, last versions. But we'd have to mock out the gap check in TransactionStreamInternal

Reply:
as a workaourdn, we could probably match starting_version and endpoint_version for each transaction stream config. let me think a bit more on this

Reply:
I think if we mock `TransactionStreamStep`, it should get around this so we don't need to specify a start/end version here.

Reply:
yeah that would be nicer. dumb question: how do we inject this step into the processor?